### PR TITLE
Fix durability section

### PR DESCRIPTION
**Pull request by ieQu1**
This PR restores the meaning of terms in the documentation for the durable storage feature.  

- It fixes the mixup between "sessions" and "storages" and adds a short section explaining the difference between these two concepts.
- It removes unnecessary categorization of sessions into "persistent" and "ephemeral". Now the document simply talks about session expiry interval, which is part of MQTT standard and as such doesn't need a lengthy introduction. 
- It also replaces term "RAM session" with "Regular sessions" to avoid any terminology overlap with the future "RAM cache" feature. "Regular sessions" is not the best term, but they are not the focus of the document, so it helps to get them out of the way. Users already familiar with EMQX should have good intuition what it means.

In summary, I removed most of unnecessary terminology to focus attention on the subject matter.

### CODE REVIEW

**Comment by ieQu1** [comment]
Moved here: https://github.com/emqx/emqx-docs/pull/2467
